axum/src/routing/path_router.rs

```diff
@@ -36,6 +36,29 @@ where
     }
 }
 
+// Validates a path in compile time, used with the vpath macro.
+// Allow dead code needed because this is only used within the vpath macro
+// and if there's no call to it, then, there's no users of it.
+#[allow(dead_code)]
+const fn validate_static_path(path: &'static str) -> &'static str {
+    if path.is_empty() {
+        panic!("Paths must start with a `/`. Use \"/\" for root routes")
+    }
+    if path.as_bytes()[0] != b'/' {
+        panic!("Paths must start with /");
+    }
+    path
+}
+
+#[macro_export]
+/// usage: .route(vpath("/")) // compiles fine
```

Comment:
You should use [documentation tests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html).